ballista/core/src/serde/mod.rs

```diff
@@ -89,31 +91,16 @@ pub struct BallistaCodec<
     physical_plan_repr: PhantomData<U>,
 }
 
-impl Default for BallistaCodec {
-    fn default() -> Self {
-        Self {
-            logical_extension_codec: Arc::new(DefaultLogicalExtensionCodec {}),
-            physical_extension_codec: Arc::new(BallistaPhysicalExtensionCodec::default()),
-            logical_plan_repr: Default::default(),
-            physical_plan_repr: Default::default(),
-        }
-    }
-}
-
 impl BallistaCodec {
-    pub fn new_with_optional_object_store(
+    pub fn new_with_object_store_and_clients(
         object_store: Option<Arc<dyn ObjectStore>>,
+        clients: Arc<Cache<String, BallistaClient>>,
```

Comment:
Can we avoid having to thread this through everywhere by just making it a 
```
static EXECUTOR_CLIENT_CACHE: OnceLock<Cache<String,BallistaClient>>
```
If we needed to make the size configurable we could just handle that by initializing it at startup

Reply:
We initializing it in DQE